Detectors/MUON/MID/Raw/src/RawBuffer.cxx

```diff
@@ -51,31 +55,42 @@ T RawBuffer<T>::next()
   return mBytes[mElementIndex++];
 }
 
+template <typename T>
+bool RawBuffer<T>::nextHeader()
+{
+  /// Goes to next RDH
+  if (mNextHeaderIndex >= mBytes.size()) {
+    // This is the end of the buffer
+    if (mUnconsumed.empty()) {
+      mElementIndex = mNextHeaderIndex;
+      mEndOfPayloadIndex = mNextHeaderIndex;
+      return false;
+    }
+    // We were reading the unconsumed part: switch to new buffer
+    mUnconsumed.clear();
+    reset();
+    mBytes = mCurrentBuffer;
+  }
+  mHeaderIndex = mNextHeaderIndex;
+  mRDH = reinterpret_cast<const header::RAWDataHeader*>(&mBytes[mHeaderIndex]);
```

Comment:
RAWDataHeader is currently an alias to RDHv4. 
I'm curious to know how will you manage the move to v5 ? The way it is now your code will only be valid for one single version of the RDH, right ? Or am I missing something ?
(not a dealbreaker for this PR, just wondering for the future...)

Reply:
AFAIK the header size does not change from v4 to v5. And I guess that it will not change even when we move to v6. If this is the case, this specific part of the code will not change.
However you're right that for the moment I only cope with v4. So, I would probably need to modify the code when moving to v5. I think that the main change is the fact that the way to access the bc and orbit will change from v4 to v5. But then again, I hope it will not change any more. I'd say that I keep the code as it is and perform the needed changes when we move to v5....